### ISSUE TITLE
Search (`C-c C-z`) panics on pattern synonyms

### ISSUE THREAD

**Issue by gallais** [agda/agda#2381]
In the presence of pattern synonyms in the context, the search functionality (`C-c C-z`) panics. E.g. looking for "Z" triggers:

> Panic: Unbound name: searchPattern.O [0,8]@90483431

````agda
module searchPattern where

data Nat : Set where
  Z : Nat
  S : Nat → Nat

pattern O = S Z
````

**Comment by andreasabel** [comment]
I wonder whether your `collectNamesIn` functions are crucially different than `getDefs`, see https://github.com/agda/agda/blob/master/src/full/Agda/Syntax/Internal/Defs.hs.
If yes, please document the specifics of `collectNamesIn`.  Otherwise, please consider using getDefs.

**Comment by gallais** [comment]
I have started by cleaning up and documenting the definition of `findMentions`. It was frankly hard to read.

I have tried to replace `collectNamesIn` with `getDefs` and it does not deliver the right behaviour. The problem is that we can (and do!) lookup theorems mentioning constructors too and constructor names are ignored by `getDefs`.

However, `Agda.Syntax.Internal.Names`' `namesIn` seems to do the job nicely! So I'm going to go ahead with this one.
